src/jlapi.c

```diff
@@ -43,13 +43,28 @@ JL_DLLEXPORT void jl_init_with_image(const char *julia_home_dir,
     jl_options.julia_home = julia_home_dir;
     if (image_relative_path != NULL)
         jl_options.image_file = image_relative_path;
+    else
+        jl_options.image_file = jl_get_default_sysimg_path();
     julia_init(JL_IMAGE_JULIA_HOME);
     jl_exception_clear();
 }
 
-JL_DLLEXPORT void jl_init(const char *julia_home_dir)
+JL_DLLEXPORT void jl_init(void)
 {
-    jl_init_with_image(julia_home_dir, NULL);
+    char *libjldir = NULL;
+
+    void *hdl = (void*)jl_load_dynamic_library_e(jl_is_debugbuild() ? "libjulia-debug" : "libjulia", JL_RTLD_DEFAULT);
```

Comment:
Probably should be NULL rather than the library name, to get the `this branch returns handle of libjulia` that we want from here (thought I had changed this):

https://github.com/JuliaLang/julia/blob/d5d728024f6b1298cd19f180183431042b754c48/src/dlload.c#L131-L145